internal/cluster/validator.go

```diff
@@ -600,3 +601,16 @@ func (v *clusterValidator) skipNetworkHostPrefixCheck(c *clusterPreprocessContex
 	}
 	return false
 }
+
+// getHostPrefixForCidrValidation returns the prefix to use when validating CIDR.
+func getHostPrefixForCidrValidation(skipHostPrefix bool, clusterNetwork *models.ClusterNetwork) int {
+	hostPrefix := int(clusterNetwork.HostPrefix)
+	if skipHostPrefix {
+		if network.IsIPv6CIDR(string(clusterNetwork.Cidr)) {
+			hostPrefix = DefaultIPV6HostPrefix
```

Comment:
Does `clusterNetwork.HostPrefix` have a meaning for your case? 
Is there any reason not to set it somewhere (not necessarily here) to the default-values, instead of overriding them just for validation?

Reply:
The hostPrefix isn't necessary in the case of a user defining a networkType other than OVN/SDN. The check is in here since it only affects validation, the networkType can be retrieved, and the CIDRvalidation still needs to run using a default prefix. That said, I'm fine with moving the setting of the hostprefix elsewhere but would like to handle that in a separate PR. This PR is only to fix the problem with the IPv6 prefix which is preventing customers from using networkTypes other than OVN/SDN in a IPv6 or dual-stack setup.